src/request/change-passphrase/ChangePassphrase.js

```diff
@@ -0,0 +1,158 @@
+/* global Nimiq */
+/* global PassphraseBox */
+/* global PassphraseSetterBox */
+/* global KeyStore */
+
+class ChangePassphrase {
+    /**
+     * If a complete page is missing it will be created.
+     * However these pages wil be the default pages which usually don't match the applications requirements.
+     * Refer to the corresponsing _build() to see the general Structure.
+     * @param {ParsedSimpleRequest} request
+     * @param {Function} resolve
+     * @param {Function} reject
+     */
+    constructor(request, resolve, reject) {
+        this._resolve = resolve;
+        this._request = request;
+        this._reject = reject;
+        /** @type {Key | null} */
+        this._key = null;
+        this._passphrase = '';
+
+        this._create();
+    }
+
+    run() {
+        /** @type {PassphraseBox} */ (this._enterPassphraseBox).reset();
+        window.location.hash = ChangePassphrase.Pages.ENTER_PASSPHRASE;
+        /** @type {PassphraseBox} */ (this._enterPassphraseBox).focus();
+    }
+
+    _create() {
+        /** @type {HTMLElement} */
+        const $enterPassphrasePage = document.getElementById(ChangePassphrase.Pages.ENTER_PASSPHRASE)
+                                  || this._buildEnterPassphrasePage();
+        /** @type {HTMLElement} */
+        const $setPassphrasePage = document.getElementById(ChangePassphrase.Pages.SET_PASSPHRASE)
+                                || this._buildSetPassphrasePage();
+
+        /** @type {HTMLFormElement} */
+        const $enterPassphraseBox = ($enterPassphrasePage.querySelector('.passphrase-box'));
+        /** @type {HTMLFormElement} */
+        const $setPassphraseeBox = ($setPassphrasePage.querySelector('.passphrase-box'));
+
+        this._enterPassphraseBox = new PassphraseBox(
+            $enterPassphraseBox, {
+                buttonI18nTag: 'passphrasebox-continue',
+                hideInput: !this._request.keyInfo.encrypted,
+            },
+        );
+        this._setPassphraseBox = new PassphraseSetterBox($setPassphraseeBox);
+
+        this._enterPassphraseBox.on(PassphraseBox.Events.CANCEL, () => this._reject(new Error('CANCEL')));
+        this._enterPassphraseBox.on(PassphraseBox.Events.SUBMIT, async phrase => {
+            document.body.classList.add('loading');
+            try {
+                const passphrase = phrase ? Nimiq.BufferUtils.fromAscii(phrase) : undefined;
+                const key = await KeyStore.instance.get(this._request.keyInfo.id, passphrase);
+                if (!key) {
+                    this._reject(new Error('No key'));
+                }
+                this._key = key;
+                /** @type {PassphraseSetterBox} */ (this._setPassphraseBox).reset();
+                window.location.hash = ChangePassphrase.Pages.SET_PASSPHRASE;
+                /** @type {PassphraseSetterBox} */ (this._setPassphraseBox).focus();
```

Comment:
If the code from the `_create()` function was to be moved into the `constructor`, then you wouldn't need all the type annotation, because the variables would not be able to be `undefined`.